### ISSUE TITLE
Random Crashes using the /random command

### ISSUE THREAD

**Issue by dangreenlee** [GridLAN/RedditBot#2]
To re-create the issue:
- Add any number of subreddits to the list
- Use /random several times
- Bot will stop responding and has crashed.

Error:
```
panic: runtime error: index out of range [0] with length 0

goroutine 61 [running]:
main.glob..func1(0x12ae440, 0xc0000b80a0)
        /Users/daniel.greenlee/Documents/GitHub/RedditBot/main.go:483 +0x4cc
main.init.2.func1(0xc00007e600, 0xc0000b80a0)
        /Users/daniel.greenlee/Documents/GitHub/RedditBot/main.go:608 +0x112
github.com/bwmarrin/discordgo.interactionCreateEventHandler.Handle(0xc00045b7d0, 0x11f2d0a, {0x12b7f60, 0xc0000b80a0})
        /Users/daniel.greenlee/go/pkg/mod/github.com/bwmarrin/discordgo@v0.23.3-0.20211204170245-092735083ddf/eventhandlers.go:478 +0x38
created by github.com/bwmarrin/discordgo.(*Session).handle
        /Users/daniel.greenlee/go/pkg/mod/github.com/bwmarrin/discordgo@v0.23.3-0.20211204170245-092735083ddf/event.go:171 +0x172
exit status 2
```

I believe this has to do with how the `getJson` function handles a bad response from the reddit api. 

Ideally, we would retry to get a successful request.

**Comment by dangreenlee** [comment]
So I don't think the issue is the getJson function. I setup this test program to try and saturate the reddit api but did not receive any failures:
```
func main() {
	// run the getJson function 5 times
	for i := 0; i < 30; i++ {
		// sleep one second
		time.Sleep(100 * time.Millisecond)
		var msg string
		// If the []subreddits is empty, it will return an error
		if len(Subreddits) == 0 {
			msg = "There are no subreddits on any list."
		} else {
			// Seed the random number generator & get a random index from the slice
			rand.Seed(time.Now().UnixNano())
			randIndex := rand.Intn(len(Subreddits))

			// Query the API for a random post from a random subreddit
			randomRedditPost := RedditPost{}
			getJson("https://reddit.com/r/"+Subreddits[randIndex]+"/random.json", &randomRedditPost)

			// Set the message to the title and url of the post
			msg = fmt.Sprintf(randomRedditPost[0].Data.Children[0].Data.Title + "\n" + randomRedditPost[0].Data.Children[0].Data.URL)
		}
		fmt.Println(msg)
	}
}
```

**Comment by dangreenlee** [comment]
I'm unable to recreate this problem in production now. I don't see an outage from the Reddit status page at all.

**Comment by dangreenlee** [comment]
Added error checking to the `json.Unmarshal` function within the `getJson` function and now see this error:
```
json: cannot unmarshal number into Go struct field .data.children.data.edited of type bool
```

**Comment by dangreenlee** [comment]
Resolved with #5
